frontend/src/pages/notebookController/SpawnerPage.tsx

```diff
@@ -277,6 +276,65 @@ const SpawnerPage: React.FC<SpawnerPageProps> = React.memo(({ setStartModalShown
     setVariableRows(updatedRows);
   };
 
+  // update the config map and secret file on the cluster
+  const checkEnvVarFile = async () => {
+    const envVarFileName = generateEnvVarFileNameFromUsername(username);
+    const envVars: EnvVarReducedTypeKeyValues = variableRows.reduce(
+      (prev, curr) => {
+        const vars: Record<string, string | number> = {};
+        const secretVars: Record<string, string | number> = {};
+        curr.variables.forEach((variable) => {
+          if (variable.type === 'text') {
+            vars[variable.name] = variable.value;
+          } else {
+            secretVars[variable.name] = variable.value;
+          }
+        });
+        return {
+          configMap: { ...prev.configMap, ...vars },
+          secrets: { ...prev.secrets, ...secretVars },
+        };
+      },
+      { configMap: {}, secrets: {} },
+    );
```

Comment:
Do we not want this as a stand alone function and provide it to `checkEnvVarFile` and also use it in createNotebook... looks as if you're doing too many things in `checkEnvVarFile` function to just in the end return the value you want for the notebook.